infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -374,4 +464,41 @@ public void shutdown() {
     public void enforceSnapshotSync(DiscoveryServiceEvent event) {
         replicationManager.enforceSnapshotSync(event);
     }
+
+    /**
+     * Check if the local cluster is a connection receiver for any of the discovered sessions
+     *
+     * @return true if the local cluster is a connection receiver for any session
+     */
+    public boolean isConnectionReceiver() {
+        Set<String> connectionEndpoints = topology.getRemoteClusterEndpoints().keySet();
+        for(LogReplicationSession session : sessions) {
+            if (!connectionEndpoints.contains(session.getSinkClusterId()) &&
+                    !connectionEndpoints.contains(session.getSourceClusterId())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * set leadership status in
+     * metadata manager : so that non-leader nodes don't overwrite updates on topology change notification
+     * LogReplicationServer : so that messages received can be appropriately handled.
+     * @param isLeader
+     */
+    public void setLeadership(boolean isLeader) {
+        metadataManager.setLeadership(isLeader);
+        router.getMsgHandler().setLeadership(isLeader);
+    }
+
+    /**
+     * Check if a session will start a connection to remote, or expect a connection from remote.
```

Comment:
this comment is a bit difficult to understand.  Can be reworded to something like 'check if this node is the connection starter for the given session'.  Also, the method name can be something like 'isConnectionStarterForSession'